src/LinkDotNet.NCronJob/Configuration/NCronJobOptionBuilder.cs

```diff
@@ -116,4 +118,61 @@ public NCronJobOptionBuilder<TJob> AddNotificationHandler<TJobNotificationHandle
         Services.TryAddScoped<IJobNotificationHandler<TJob>, TJobNotificationHandler>();
         return this;
     }
+
+    /// <summary>
+    /// Adds a job that runs after the given job has finished.
+    /// </summary>
+    /// <param name="success">Configure a job that runs after the principal job has finished successfully.</param>
+    /// <param name="faulted">Configure a job that runs after the principal job has faulted. Faulted means that the parent job did throw an exception.</param>
+    /// <returns>The builder to add more jobs.</returns>
+    public NCronJobOptionBuilder<TJob> When(Action<DependencyBuilder<TJob>>? success = null, Action<DependencyBuilder<TJob>>? faulted = null)
+    {
+        if (success is not null)
+        {
+            var dependencyBuilder = new DependencyBuilder<TJob>(true);
+            success(dependencyBuilder);
+            dependencyBuilder.GetDependentJobOption().ForEach(s =>
+            {
+                Services.AddSingleton(s);
+                Services.TryAddSingleton(s.DependentJobType);
```

Comment:
I do think we don't want to force needless registration and therefore we do it automatically here.